### PR TITLE
Development

### PR DESCRIPTION
**Pull request by asad451**
- Cater previous changes
- Extra attributes are added in devise gem
- "Edit Profile" functionality is added
- Admin is added using enum
- Admin Workspace is created which shows all registered users

### CODE REVIEW

File: app/assets/stylesheets/style.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
+.username{
```
Comment (taimur-akhtar) [inline app/assets/stylesheets/style.scss]: give space above this line.

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.container {
+    width: 40%;
```
Comment (taimur-akhtar) [inline app/assets/stylesheets/users.scss]: spacing.

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    margin: 0 auto;
+}
+
+.form-control{
+  width: 70%;
```
Comment (taimur-akhtar) [inline app/assets/stylesheets/users.scss]: spacing.

File: app/assets/stylesheets/users.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  margin: 0 auto
+}
+.label-space{
+  margin-left: 15%;
```
Comment (taimur-akhtar) [inline app/assets/stylesheets/users.scss]: spacing.

File: app/assets/stylesheets/users.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+}
+.btn-space{
+  margin-left: 39%;
```
Comment (taimur-akhtar) [inline app/assets/stylesheets/users.scss]: indentation.

File: app/controllers/admin/admins_controller.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class Admin::AdminsController < ApplicationController
+  def index
```
Comment (taimur-akhtar) [inline app/controllers/admin/admins_controller.rb]: no authentication for whether user is an admin or not?

File: app/controllers/home_controller.rb [hunk 7]
```diff
@@ -1,4 +1,6 @@
-class HomeController < ApplicationController
+class HomeController < ApplicationController
```
Comment (taimur-akhtar) [inline app/controllers/home_controller.rb]: why is there a space in the end? remove that please

File: app/views/admin/admins/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    </thead>
+    <tbody>
+      <% @users.each do |user| %>
+        <tr>
+            <td><%= user.username %></td>
+            <td><%= user.email %></td>
+            <td><%= user.contact %></td>
+            <td><%= user.gender %></td>
+            <td><%= user.role %></td>
+            <td><button class="btn btn-info">Enable</button>    <button class="btn btn-info">Disable</button></td>
```
Comment (taimur-akhtar) [inline app/views/admin/admins/index.html.erb]: lets discuss this part tomorrow.

File: app/views/home/index.html.erb [hunk 9]
```diff
@@ -1,2 +1 @@
-<h1>Home#index</h1>
-<p>Find me in app/views/home/index.html.erb</p>
+<h1 align="center">Project Management System</h1>
```
Comment (taimur-akhtar) [inline app/views/home/index.html.erb]: use classes for styling.

File: app/views/shared/_header.html.erb [hunk 10]
```diff
@@ -0,0 +1,52 @@
+<div class="container-fluid">
+  <nav class="navbar navbar-inverse">
+    <ul class="nav navbar-nav navbar-left">
+      <li><%= link_to "PMS", root_path %></li>
+      <% if user_signed_in? && current_user.role == "admin" %>
```
Comment (taimur-akhtar) [inline app/views/shared/_header.html.erb]: move this condition to a helper and you should be calling a method like if admin_user?

Comment (taimur-akhtar) [reply app/views/shared/_header.html.erb]: and please read the documentation of enum....why are we using == 'admin'

File: app/views/shared/_header.html.erb [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+            <%= link_to destroy_user_session_path, method: :delete do %>
+              <span class="glyphicon glyphicon-log-out"></span>   Log Out
+            <% end %>
+          </li>
+          <li> 
+            <%= link_to edit_user_registration_path do %>
+              <span class="glyphicon glyphicon-edit"></span>   Edit Profile
+            <% end %>
+          </li>
+
```
Comment (taimur-akhtar) [inline app/views/shared/_header.html.erb]: remove this line.

File: config/routes.rb [hunk 12]
```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
+
   devise_for :users
   root to: "home#index"
-
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  
+  namespace :admin do
+    root to: 'admins#index'
```
Comment (taimur-akhtar) [inline config/routes.rb]: keep root one in the end.

File: config/routes.rb [hunk 13]
```diff
@@ -1,6 +1,11 @@
...
+    resources :users
```
Comment (taimur-akhtar) [inline config/routes.rb]: where are these being used?

File: test/controllers/admin/admins_controller_test.rb [hunk 14]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
```
Comment (taimur-akhtar) [inline test/controllers/admin/admins_controller_test.rb]: no need for these test files, 
rm them when they are generated.

File: app/assets/stylesheets/style.scss (hunk 1 above)
Comment (asad451) [reply app/assets/stylesheets/style.scss]: fixed

File: app/assets/stylesheets/users.scss (hunk 2 above)
Comment (asad451) [reply app/assets/stylesheets/users.scss]: fixed

File: app/assets/stylesheets/users.scss (hunk 4 above)
Comment (asad451) [reply app/assets/stylesheets/users.scss]: fixed

File: app/assets/stylesheets/users.scss (hunk 5 above)
Comment (asad451) [reply app/assets/stylesheets/users.scss]: fixed

File: app/controllers/home_controller.rb (hunk 7 above)
Comment (asad451) [reply app/controllers/home_controller.rb]: fixed

File: app/views/home/index.html.erb (hunk 9 above)
Comment (asad451) [reply app/views/home/index.html.erb]: fixed

File: app/views/shared/_header.html.erb (hunk 11 above)
Comment (asad451) [reply app/views/shared/_header.html.erb]: fixed

File: app/assets/stylesheets/users.scss (hunk 3 above)
Comment (asad451) [reply app/assets/stylesheets/users.scss]: fixed

File: app/controllers/admin/admins_controller.rb (hunk 6 above)
Comment (asad451) [reply app/controllers/admin/admins_controller.rb]: fixed

**Pull request merged**